src/waitForPipeline.ts

```
@@ -0,0 +1,59 @@
+import {Execution, Statuses} from './main'
+import {AxiosInstance} from 'axios'
+
+export async function WaitUntilPipelineCompleteOrTimeout(
+  url: string,
+  eventId: string,
+  timeout: number,
+  sleep: number,
+  initialTime: Date,
+  client: AxiosInstance
+): Promise<string> {
+  if (new Date().getTime() - initialTime.getTime() <= timeout) {
+    try {
+      const response = await client.get<Execution[]>(url, {params: {eventId}})
+      if (
+        response.data.length !== 0 &&
+        isPipelineCompleted(response.data[0].status)
+      ) {
+        return Promise.resolve(response.data[0].status)
+      }
+      await delay(sleep)
+      return WaitUntilPipelineCompleteOrTimeout(
+        url,
+        eventId,
+        timeout,
+        sleep,
+        initialTime,
+        client
+      )
+    } catch (e) {
```

I think in the newest TS/JS versions you can just write 
```
} catch {
...
}
```

if you're going to ignore the error.